app/controllers/facility_accounts_controller.rb

```diff
@@ -70,7 +47,31 @@ def show
   # GET /facilities/:facility_id/accounts/new
   def new
     @owner_user = User.find(params[:owner_user_id])
-    @account    = @owner_user.accounts.new(:expires_at => Time.zone.now + 1.year)
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+    @account = current_account_type.constantize.new
+  end
+
+  # POST /facilities/:facility_id/accounts
+  def create
+    @owner_user = User.find(params[:owner_user_id])
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+
+    @account = AccountBuilder.for(params[:account_type]).new({
+      account_type: params[:account_type],
+      facility: current_facility,
+      current_user: current_user,
+      owner_user: @owner_user,
+      params: params,
+    }).build
+
+    if @account.save
+      flash[:notice] = 'Account was successfully created.'
+      redirect_to facility_user_accounts_path(current_facility, @account.owner_user)
+    else
+      render action: 'new'
```

Comment:
Should we have a `flash.now` or do the errors on the form itself suffice? (haven't done a testing yet)